x/mongo/driver/topology/connection.go

```diff
@@ -111,17 +111,17 @@ func (c *connection) connect(ctx context.Context) {
 	}
 	defer close(c.connectDone)
 
-	c.connectContextMutex.Lock()
+	c.mu.Lock()
```

Comment:
I can revert these changes as well. I do not believe this variable is referenced outside of the driver code so there should be no impact